lib/IRGen/GenClass.cpp

```diff
@@ -261,13 +261,17 @@ namespace {
                ClassHasObjCAncestry);
     }
 
-    bool doesMetadataRequireDynamicInitialization() const {
+    bool doesMetadataRequireInitialization() const {
       return (ClassHasMissingMembers ||
               ClassHasResilientMembers ||
               ClassHasResilientAncestry ||
               ClassHasGenericAncestry);
     }
 
+    bool doesMetadataRequireRelocation() const {
+      return ClassHasResilientAncestry;
```

Comment:
Your commit comment implies that the class itself being generic would affect this too. Does the question just not get asked in that case?

Reply:
Generic classes go down a totally separate code path. Note that the top-level irgen::doesClassMetadataRequireRelocation() method does return true for generic classes. I can add a comment here.